### ISSUE TITLE
Missing Doc: Array.prototype.join calls toString internally

### ISSUE THREAD

**Issue by amustaque97** [mdn/content#27134]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join

### What specific section or headline is this issue about?

Examples

### What information was incorrect, unhelpful, or incomplete?

We should demonstrate an example where we tell the readers that if there are nested arrays as items of an array variable If we call the `join` method on the variable it will internally call `toString` on the nested array item with `,` as a delimiter. This will return a flattering array ( which may confuse the users if their delimiter is  also comma `,` ), for example below

**Code snippet**
```javascript
const x = ['h', 'e', 'l', 'l', 'o'];

console.log([[[[[[[[x]]]]]]], 'w', 'o', 'r', 'l', 'd'].join(','))
```

**output**:
```shell
"h,e,l,l,o,w,o,r,l,d" 
```

### What did you expect to see?

Examples like this can be really helpful if someone is a beginner and it can help people during dev or code reviews.

### Do you have any supporting links, references, or citations?

Under the spec: 
https://tc39.es/ecma262/multipage/indexed-collections.html#sec-array.prototype.join

Read point `7.b`

### Do you have anything more you want to share?

If we all agree on this, I'm happy to raise an MR.

**Comment by Josh-Cena** [comment]
I'm not sure what you could add. The fact that "Array.prototype.join calls toString internally" is mentioned on both https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toString. What behavior should it have otherwise, anyway?

**Comment by amustaque97** [comment]
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toString

I agree `toString` page mentions calling the `join` method but I'm not able to find that line in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join  `join` says it calls `toString` internally though it only says "new string by concatenating all of the elements in an array" but how it happens we don't tell the reader.

Thoughts?

**Comment by Josh-Cena** [comment]
Ummm... I see your point. join and toString are mutually recursive. Sure, we can add a sentence or a few examples.

Fun fact: when there's a mutually recursive call you naturally think about what happens when there is a loop:

```js
const a = [];
a[0] = a;
a.join();
```

And, you will find that none of the browsers get stuck! See https://github.com/tc39/ecma262/issues/289. Before the issue is closed, I don't think we have to mention this edge case; just some food for thought.
